lib/rules/no-unused-expressions.js

```diff
@@ -129,12 +129,37 @@ module.exports = {
 			/**
 			 * https://tc39.es/ecma262/#directive-prologue
 			 *
-			 * Only `FunctionBody`, `ScriptBody` and `ModuleBody` can have directive prologue.
-			 * Class static blocks do not have directive prologue.
+			 * Only `FunctionBody`, `ScriptBody`, `ModuleBody`, and TypeScript module/namespace bodies
+			 * can have directive prologue. Class static blocks do not have directive prologue.
 			 */
+			if (!looksLikeDirective(node)) {
+				return false;
+			}
```

Comment:
[Refactor] Deleting this `if` altogether doesn't cause any unit tests to fail. Looks to me like either it's unnecessary or there are some missing unit tests. WDYT?

```suggestion
```